dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -196,4 +140,65 @@ private string DebuggerDisplay
             return display;
         }
     }
+
+    #region Error handling
+    /// <summary>
+    /// Whether an error occurred while executing functions in the pipeline.
+    /// </summary>
+    public bool ErrorOccurred { get; private set; }
+
+    /// <summary>
+    /// Error details.
+    /// </summary>
+    public string LastErrorDescription { get; private set; } = string.Empty;
```

Comment:
Should this be nullable?

Reply:
Good call. Keeping this PR focused on the indexer[] (this part of the diff is just moving the exception-related stuff around within the file). Making it nullable does trigger a few warnings scattered in files throughout the repo, so I'll make that one separately.

Reply:
Created issue #2193 to track. I'll try to get that one out today.